Website/app_code/code/Blog.cs

```diff
@@ -63,10 +63,10 @@ public static Post CurrentPost
         {
             if (HttpContext.Current.Items["currentpost"] == null && !string.IsNullOrEmpty(CurrentSlug))
             {
-                var post = Storage.GetAllPosts().FirstOrDefault(p => p.Slug == CurrentSlug);
+                var post = GetVisiblePosts().FirstOrDefault(p => p.Slug == CurrentSlug);
 
-                if (post != null && (post.IsPublished || HttpContext.Current.User.Identity.IsAuthenticated || (HttpContext.Current.Request.QueryString["key"] ?? string.Empty).Equals(post.ID, StringComparison.InvariantCultureIgnoreCase)))
```

Comment:
There was a conditional statement to include a post of its id is specified as a query string parameter called 'key' however I couldn't find a usage of that and it's not clear what scenario this was supposed to cover. Authenticated users will be able to see all posts and non-authenticated users will see posts that are published and their pubDates are in the past.